aiida/restapi/translator/node.py

```diff
@@ -606,78 +607,125 @@ def get_node_shape(ntype):
         # get all inputs
         qb_obj = QueryBuilder()
         qb_obj.append(Node, tag="main", project=['*'], filters=self._id_filter)
-        qb_obj.append(Node, tag="in", project=['*'], edge_project=['label'], with_outgoing='main')
+        qb_obj.append(Node, tag="in", project=['*'], edge_project=['label', 'type'], with_outgoing='main')
+        if tree_in_limit is not None:
+            qb_obj.limit(tree_in_limit)
 
-        if qb_obj.count() > 0:
+        input_node_pks = {}
+        sent_no_of_incomings = qb_obj.count()
+
+        if sent_no_of_incomings > 0:
             for node_input in qb_obj.iterdict():
                 node = node_input['in']['*']
-                linktype = node_input['main--in']['label']
+                # add node if it is not present
```

Comment:
should this be moved a bit down before the "if" statement?

Reply:
done